crates/sui-framework/packages/sui-framework/tests/random_tests.move

```diff
@@ -216,6 +222,9 @@ module sui::random_tests {
             scenario.ctx(),
         );
 
+        // Deterministically set tx context so the RNG seeded by the ctx below is deterministic.
+        *scenario.ctx() = tx_context::new_from_hint(@0x0, 1, 0, 0, 0);
+
```

Comment:
I am afraid

Reply:
When the `Rng` is generated, it takes the source of randomness that is shared between transactions in a checkpoint (`Random`) and mixes it with its own transaction digest (by generating a new address and then using that), so if the digest is not stable in this case, you'll get different random numbers each time.

Reply:
My fear was from writing to `scenario.ctx()`, not the hard coding of the context itself